topics/data-science/tutorials/python-typing/tutorial.md

```diff
@@ -0,0 +1,251 @@
+---
+layout: tutorial_hands_on
+
+title: Python - Typing
+level: Intermediate
+requirements: []
+follow_up_training: []
+questions:
+- What is typing?
+- How does it improve code?
+- Can it help me?
+
+objectives:
+- Understand the utility of annotating types on one's code
+- Understand the limits of typing in python
+
+time_estimation:  30M
+key_points:
+- Typing improves the correctness and quality of your code
+- It can ensure that editor provided hints are better and more accurate.
+
+subtopic: python-modular
+contributions:
+  authorship:
+  - hexylena
+  editing:
+  - dirowa
+  - bazante1
+
+priority: 3
+notebook:
+  language: python
+---
+
+In some, better, languages typing is a core part of the language and types are checked at compile time, to ensure your code can never use the incorrect type of object. Python, and a few other dynamic languages, instead use "Duck Typing" wherein the type of the object is less important than whether or not the correct methods or attributes are available.
```

Comment:
Is the "better" here necessary?

Reply:
better is indeed not necessary and probably too colourful (but I'm increasingly of the mind that strongly typed languages are the right answer in general, I'll try and cut the personality out.)

linking to duck typing is a good idea! I wrote this on the train with minimal internet, hence complete lack of links to useful documentation/etc.

Reply:
done!